core/src/main/java/org/apache/accumulo/core/rpc/clients/ShellAuthenticatorThriftClient.java

```diff
@@ -58,16 +58,26 @@ public Pair<String,Client> getTabletServerConnection(Logger LOG, ThriftClientTyp
     checkArgument(context != null, "context is null");
     final long rpcTimeout = context.getClientTimeoutInMillis();
 
-    List<String> tservers = new ArrayList<>();
     final ZooCache zc = context.getZooCache();
-    for (String tserver : zc.getChildren(context.getZooKeeperRoot() + Constants.ZTSERVERS)) {
-      tservers.add(tserver);
-    }
-    if (tservers.isEmpty() && !warnedAboutTServersBeingDown.get()) {
-      LOG.warn("There are no tablet servers: check that zookeeper and accumulo are running.");
-      warnedAboutTServersBeingDown.set(true);
-      Collections.shuffle(tservers);
+    final List<String> tservers = new ArrayList<>();
+
+    for (int retries = 0; retries < 10; retries++) {
+      // Cluster may not be up, wait for tservers to come online
+      while (true) {
+        for (String tserver : zc.getChildren(context.getZooKeeperRoot() + Constants.ZTSERVERS)) {
+          tservers.add(tserver);
+        }
+        if (!tservers.isEmpty()) {
+          break;
+        }
+        if (!tservers.isEmpty() && !warnedAboutTServersBeingDown.get()) {
+          LOG.warn("There are no tablet servers: check that zookeeper and accumulo are running.");
+          warnedAboutTServersBeingDown.set(true);
+        }
+      }
```

Comment:
Would there be a benefit of some delay between retries, thinking along the lines of 50ms or so so that ZooKeeper has a change to breath between getChildren() attempts?

Reply:
Fixed in b2d17a2